doc/src/modules/physics/vector/advanced.rst

```diff
@@ -8,6 +8,48 @@ the features that will be implemented in the future will also be covered, along
 with unanswered questions about proper functionality. Also, common problems
 will be discussed, along with some solutions.
 
+Time Variable
+=============
+
+The objects in `sympy.physics.vector` and `sympy.physics.mechanics` rely on a
+default symbol for time. If symbols are created with ``dynamicsymbols()`` they
+are implicit functions of the default time variable. This default symbol can be
+accessed like so::
+
+   >>> from sympy.physics.vector import TIME
+
+The default behavior is::
+
+   >>> TIME
+   t
+   >>> from sympy.physics.vector import dynamicsymbols
+   >>> q = dynamicsymbols('q')
+   >>> q
+   q(t)
+   >>> list(q.free_symbols)[0] is TIME
+   True
+   >>> q.diff(TIME)
+   Derivative(q(t), t)
+
+There are sometimes cases where you may want to set the default variable for
+time to something other than the default. It is possible to do so, but it
+requires monkey patching the variable in the module. **Note that this is
+fragile and you can likely encounter errors if the imports are not carefully
+done.** You can set a different symbol for time like so::
```

Comment:
Just looking at the code below, this isn't going to work. If you want it to work, you have to replace 

```py
from sympy.physics.vector.functions import TIME

...

TIME
```

with 

```py
from sympy.physics.vector import functions

...

functions.TIME
```

Otherwise `TIME` will still be what it was imported as. If you want to support this, you need to make this basically the *only* supported way of accessing this API. It needs to work like that in non-library code as well, if that code is expected to support changing the time variable. 

Personally, I would recommend against supporting this. A much simpler way to get the same functionality would be to use a custom object for TIME and allow modifying the printer for it.

Reply:
Ironically, the old method, `dynamicsymbols._t`, actually does support this, because `_t` is used as an attribute. At best, if you want this to work this way, you should make it an attribute of an object rather than a top level name, to make it impossible to import it in a way that breaks monkeypatching (it would still break if someone did `t = dynamicsymbols.TIME`, but there's no way to avoid that while keeping with the immutable design, which is why I highly recommend doing the printer approach instead).

Reply:
> A much more robust way to get the same functionality would be to use a custom object for TIME and allow modifying the printer for it.

This is worth some thought. Thanks.